src/main/java/View/playlistView.java

```diff
@@ -2,16 +2,17 @@
 
 import Entities.User;
 import ViewModel.musicEngineControllerViewModel;
-import ViewModel.spaceViewModel;
 
 import javax.swing.*;
 import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
 
 /**
  * view for the user dashboard
  */
-public class playlistView extends JFrame {
-    private final User user;
+public class playlistView extends JFrame implements ActionListener {
+    private User user;
```

Comment:
I know you mentioned this (use of raw User type in view) in the pull comment, we could even raise an issue for this since it's a general issue.

Reply:
will do